Add the pre-generated JNI header for the adbc-driver-jni bridge.

Upstream generates this header by running Maven over the Java sources
(java/driver/jni/CMakeLists.txt calls `mvn -Pjni,javah compile`). Doing that
here would make build-thirdparty.sh the first package to require Maven, a
Maven Central connection and a JDK 11+ -- the official build-env image runs
this script with JAVA_HOME pointing at JDK 8 (docker/compilation/Dockerfile).

The header is pure javah output and is fully determined by the pinned version
of java/driver/jni/src/main/java/org/apache/arrow/adbc/driver/jni/impl/NativeAdbc.java,
so it is checked in instead. build_arrow_adbc() compiles jni_wrapper.cc against
it directly, needing nothing but jni.h.

To regenerate when bumping arrow-adbc, run inside the extracted source tree:

    cmake -S java -B /tmp/adbcjni -DCMAKE_PREFIX_PATH="${TP_INSTALL_DIR}"
    cmake --build /tmp/adbcjni
    cp java/driver/jni/target/headers/org_apache_arrow_adbc_driver_jni_impl_NativeAdbc.h \
       java/driver/jni/doris_generated/

then re-diff this file.

diff -uNr a/java/driver/jni/doris_generated/org_apache_arrow_adbc_driver_jni_impl_NativeAdbc.h b/java/driver/jni/doris_generated/org_apache_arrow_adbc_driver_jni_impl_NativeAdbc.h
--- a/java/driver/jni/doris_generated/org_apache_arrow_adbc_driver_jni_impl_NativeAdbc.h	1970-01-01 08:00:00.000000000 +0800
+++ b/java/driver/jni/doris_generated/org_apache_arrow_adbc_driver_jni_impl_NativeAdbc.h	2026-07-31 14:45:35.581346015 +0800
@@ -0,0 +1,429 @@
+/* DO NOT EDIT THIS FILE - it is machine generated */
+#include <jni.h>
+/* Header for class org_apache_arrow_adbc_driver_jni_impl_NativeAdbc */
+
+#ifndef _Included_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+#define _Included_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+#ifdef __cplusplus
+extern "C" {
+#endif
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    openDatabase
+ * Signature: (I[[B)Lorg/apache/arrow/adbc/driver/jni/impl/NativeDatabaseHandle;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_openDatabase
+  (JNIEnv *, jclass, jint, jobjectArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    closeDatabase
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_closeDatabase
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    openConnection
+ * Signature: (J)Lorg/apache/arrow/adbc/driver/jni/impl/NativeConnectionHandle;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_openConnection
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    closeConnection
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_closeConnection
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    openStatement
+ * Signature: (J)Lorg/apache/arrow/adbc/driver/jni/impl/NativeStatementHandle;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_openStatement
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    closeStatement
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_closeStatement
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementBind
+ * Signature: (JJJ)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementBind
+  (JNIEnv *, jclass, jlong, jlong, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementBindStream
+ * Signature: (JJ)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementBindStream
+  (JNIEnv *, jclass, jlong, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementCancel
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementCancel
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementExecuteUpdate
+ * Signature: (J)J
+ */
+JNIEXPORT jlong JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementExecuteUpdate
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementPrepare
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementPrepare
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementExecutePartitions
+ * Signature: (J)Lorg/apache/arrow/adbc/driver/jni/impl/NativePartitionResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementExecutePartitions
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementExecuteQuery
+ * Signature: (J)Lorg/apache/arrow/adbc/driver/jni/impl/NativeQueryResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementExecuteQuery
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementExecuteSchema
+ * Signature: (J)Lorg/apache/arrow/adbc/driver/jni/impl/NativeSchemaResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementExecuteSchema
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementGetParameterSchema
+ * Signature: (J)Lorg/apache/arrow/adbc/driver/jni/impl/NativeSchemaResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementGetParameterSchema
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementSetSqlQuery
+ * Signature: (J[B)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementSetSqlQuery
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementSetSubstraitPlan
+ * Signature: (JLjava/nio/ByteBuffer;)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementSetSubstraitPlan
+  (JNIEnv *, jclass, jlong, jobject);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementGetOptionBytes
+ * Signature: (J[B)[B
+ */
+JNIEXPORT jbyteArray JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementGetOptionBytes
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementGetOptionDouble
+ * Signature: (J[B)D
+ */
+JNIEXPORT jdouble JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementGetOptionDouble
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementGetOptionLong
+ * Signature: (J[B)J
+ */
+JNIEXPORT jlong JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementGetOptionLong
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementGetOptionString
+ * Signature: (J[B)[B
+ */
+JNIEXPORT jbyteArray JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementGetOptionString
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementSetOptionBytes
+ * Signature: (J[B[B)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementSetOptionBytes
+  (JNIEnv *, jclass, jlong, jbyteArray, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementSetOptionDouble
+ * Signature: (J[BD)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementSetOptionDouble
+  (JNIEnv *, jclass, jlong, jbyteArray, jdouble);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementSetOptionLong
+ * Signature: (J[BJ)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementSetOptionLong
+  (JNIEnv *, jclass, jlong, jbyteArray, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    statementSetOptionString
+ * Signature: (J[B[B)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_statementSetOptionString
+  (JNIEnv *, jclass, jlong, jbyteArray, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionCancel
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionCancel
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetObjects
+ * Signature: (JI[B[B[B[[B[B)Lorg/apache/arrow/adbc/driver/jni/impl/NativeQueryResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetObjects
+  (JNIEnv *, jclass, jlong, jint, jbyteArray, jbyteArray, jbyteArray, jobjectArray, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetInfo
+ * Signature: (J[I)Lorg/apache/arrow/adbc/driver/jni/impl/NativeQueryResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetInfo
+  (JNIEnv *, jclass, jlong, jintArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetTableSchema
+ * Signature: (J[B[B[B)Lorg/apache/arrow/adbc/driver/jni/impl/NativeSchemaResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetTableSchema
+  (JNIEnv *, jclass, jlong, jbyteArray, jbyteArray, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetTableTypes
+ * Signature: (J)Lorg/apache/arrow/adbc/driver/jni/impl/NativeQueryResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetTableTypes
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionCommit
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionCommit
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionRollback
+ * Signature: (J)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionRollback
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionReadPartition
+ * Signature: (JLjava/nio/ByteBuffer;)Lorg/apache/arrow/adbc/driver/jni/impl/NativeQueryResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionReadPartition
+  (JNIEnv *, jclass, jlong, jobject);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetOptionBytes
+ * Signature: (J[B)[B
+ */
+JNIEXPORT jbyteArray JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetOptionBytes
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetOptionDouble
+ * Signature: (J[B)D
+ */
+JNIEXPORT jdouble JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetOptionDouble
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetOptionLong
+ * Signature: (J[B)J
+ */
+JNIEXPORT jlong JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetOptionLong
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetOptionString
+ * Signature: (J[B)[B
+ */
+JNIEXPORT jbyteArray JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetOptionString
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionSetOptionBytes
+ * Signature: (J[B[B)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionSetOptionBytes
+  (JNIEnv *, jclass, jlong, jbyteArray, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionSetOptionDouble
+ * Signature: (J[BD)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionSetOptionDouble
+  (JNIEnv *, jclass, jlong, jbyteArray, jdouble);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionSetOptionLong
+ * Signature: (J[BJ)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionSetOptionLong
+  (JNIEnv *, jclass, jlong, jbyteArray, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionSetOptionString
+ * Signature: (J[B[B)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionSetOptionString
+  (JNIEnv *, jclass, jlong, jbyteArray, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetStatisticNames
+ * Signature: (J)Lorg/apache/arrow/adbc/driver/jni/impl/NativeQueryResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetStatisticNames
+  (JNIEnv *, jclass, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    connectionGetStatistics
+ * Signature: (J[B[B[BZ)Lorg/apache/arrow/adbc/driver/jni/impl/NativeQueryResult;
+ */
+JNIEXPORT jobject JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_connectionGetStatistics
+  (JNIEnv *, jclass, jlong, jbyteArray, jbyteArray, jbyteArray, jboolean);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    databaseGetOptionBytes
+ * Signature: (J[B)[B
+ */
+JNIEXPORT jbyteArray JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_databaseGetOptionBytes
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    databaseGetOptionDouble
+ * Signature: (J[B)D
+ */
+JNIEXPORT jdouble JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_databaseGetOptionDouble
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    databaseGetOptionLong
+ * Signature: (J[B)J
+ */
+JNIEXPORT jlong JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_databaseGetOptionLong
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    databaseGetOptionString
+ * Signature: (J[B)[B
+ */
+JNIEXPORT jbyteArray JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_databaseGetOptionString
+  (JNIEnv *, jclass, jlong, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    databaseSetOptionBytes
+ * Signature: (J[B[B)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_databaseSetOptionBytes
+  (JNIEnv *, jclass, jlong, jbyteArray, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    databaseSetOptionDouble
+ * Signature: (J[BD)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_databaseSetOptionDouble
+  (JNIEnv *, jclass, jlong, jbyteArray, jdouble);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    databaseSetOptionLong
+ * Signature: (J[BJ)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_databaseSetOptionLong
+  (JNIEnv *, jclass, jlong, jbyteArray, jlong);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    databaseSetOptionString
+ * Signature: (J[B[B)V
+ */
+JNIEXPORT void JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_databaseSetOptionString
+  (JNIEnv *, jclass, jlong, jbyteArray, jbyteArray);
+
+/*
+ * Class:     org_apache_arrow_adbc_driver_jni_impl_NativeAdbc
+ * Method:    internalGetByteBuffer
+ * Signature: (Ljava/nio/ByteBuffer;)[B
+ */
+JNIEXPORT jbyteArray JNICALL Java_org_apache_arrow_adbc_driver_jni_impl_NativeAdbc_internalGetByteBuffer
+  (JNIEnv *, jclass, jobject);
+
+#ifdef __cplusplus
+}
+#endif
+#endif
